transcribe_demo.py

```diff
@@ -106,22 +107,16 @@ def record_callback(_, audio:sr.AudioData) -> None:
                     phrase_complete = True
                 # This is the last time we received new audio data from the queue.
                 phrase_time = now
-
-                # Concatenate our current audio data with the latest audio data.
-                while not data_queue.empty():
-                    data = data_queue.get()
-                    last_sample += data
-
-                # Use AudioData to convert the raw data to wav data.
-                audio_data = sr.AudioData(last_sample, source.SAMPLE_RATE, source.SAMPLE_WIDTH)
-                wav_data = io.BytesIO(audio_data.get_wav_data())
-
-                # Write wav data to the temporary file as bytes.
-                with open(temp_file, 'w+b') as f:
-                    f.write(wav_data.read())
+                
+                # Combine audio data from queue
+                audio_data = b''.join(data_queue.queue)
+                data_queue.queue.clear()
+                
+                # Convert in-ram buffer to something the model can use directly without needing a temp file.
+                audio_np = np.frombuffer(audio_data, dtype=np.int16).astype(np.float32) / 32768.0
```

Comment:
Can you clarify this magic number in a comment as 2**15 or save it to a variable named `maxInt` or something to make it a little more clear what we're doing?

Reply:
Does that work?

We are working with raw PCM audio data at this point, so are right, right due to the mixed concerns the comments are useful.  I'm just of the habit of not adding extra comments if the code explains itself, so I think in this case its more required to teach others, and I'm happy to do so.